superset/models/helpers.py

```diff
@@ -1749,12 +1753,20 @@ def get_sqla_query(  # pylint: disable=too-many-arguments,too-many-locals,too-ma
                 inner_time_filter = []
 
                 if dttm_col and not db_engine_spec.time_groupby_inline:
-                    inner_time_filter = [
-                        dttm_col.get_time_filter(
-                            inner_from_dttm or from_dttm,
-                            inner_to_dttm or to_dttm,
-                        )
-                    ]
+                    if isinstance(dttm_col, dict):
```

Comment:
Use this pattern instead for grabbing the time_filter:
https://github.com/preset-io/superset/blob/master/superset/models/helpers.py#L1550